### DISCUSSION TITLE
http hook throws when `params` passed as kwarg

### DISCUSSION THREAD

**Discussion by kiaradlf** [apache/airflow#39778] [General]
### Apache Airflow version

2.9.1

### What happened?

http hook throws when `params` passed as kwarg

### What you think should happen instead?

no error

### How to reproduce

when i invoke:

```
from airflow.providers.http.hooks.http import HttpHook

HttpHook.__init__(method="GET", http_conn_id="foo").run(** {'endpoint': '/foos', 'params': {'page': 1}})
```

i get the error:

```
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/http/hooks/http.py", line 177, in run
    req = requests.Request(self.method, url, params=data, headers=headers, **request_kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: requests.models.Request() got multiple values for keyword argument 'params'
```

### Operating System

chromeos

### Versions of Apache Airflow Providers

1.9.0

### Deployment

Docker-Compose

### Anything else?

while i found this not immediately obvious, it seems to go through when this is passed in kwargs as `data` rather than `params`

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by Taragolis** [comment]
I'm not sure about version of provider `1.9.0` however according to the latest (`4.11.0` at that moment) [HttpHook.run](https://airflow.apache.org/docs/apache-airflow-providers-http/stable/_api/airflow/providers/http/hooks/http/index.html#airflow.providers.http.hooks.http.HttpHook.run) you should provide it as a `data` argument

```python
HttpHook(method="GET", http_conn_id="foo").run(endpoint='/foos', data={'page': 1})
```
